src/openms/source/FILTERING/CALIBRATION/InternalCalibration.cpp

```diff
@@ -344,31 +344,33 @@ namespace OpenMS
       tms.reserve(exp.size());
       // go through spectra and calibrate
       Size i(0), i_mslvl(0);
-      for (PeakMap::Iterator it = exp.begin(); it != exp.end(); ++it, ++i)
+      for (auto& spec : exp)
```

Comment:
we may need to rethink our policy of "replace with range-based-for at all costs".
This is an example where it made the code more error prone.
I'd suggest to either keep the old version.
There are other possibilities, but all of them involve more code. The old version is the most compact one, I can think of.
We could switch to a simple `for (Size i = 0; i < ...)` + `const auto& spec = exp[i];` if the iterator is something that is bugging someone, but I dont think it makes the code better.

Reply:
I think this is a reasonable concern. Should we roll those changes back - which means some of us needs to do it. Or give it another review to ensure that I did not miss one of these dangerous loops.